docs/docs/associations.md

```diff
@@ -126,19 +126,64 @@ Game.belongsTo(Team);
 
 Even though it is called a HasOne association, for most 1:1 relations you usually want the BelongsTo association since BelongsTo will add the foreignKey on the source where hasOne will add on the target.
 
+### Difference between HasOne and BelongsTo
+
+In Sequelize 1:1 relationship can be set using HasOne and BelongsTo. They are suitable for different scenarios. Lets study this difference using an example.
+
+Suppose we have two tables to link **Player** and **Team**. Lets define their models.
+
+```js
+var Player = this.sequelize.define('player', {/* attributes */})
+  , Team  = this.sequelize.define('team', {/* attributes */});
+```
+
+When we link two model in Sequelize we can refer them as pairs of **source** and **target** models. Like this
+
+Having **Player** as the **source** and **Team** as the **target**
+```js
+Player.belongsTo(Team);
+//Or
+Player.hasOne(Team);
+```
+
+Having **Team** as the **source** and **Player** as the **target**
+```js
+Team.belongsTo(Player);
+//Or
+Team.hasOne(Player);
+```
+
+Now depending upon where the Foreign Key relation exists we can use either HasOne or BelongsTo.
+
+If **source** model has the Foreign Key , then use BelongsTo. For example
+
+**Player** model has the `teamId` key which makes it **source** relation. We should use the BelongsTo
+
+```js
+Player.belongsTo(Team); // This will add `teamId` into Player i.e. source model
+```
+
+If **target** model has the Foreign Key , then use HasOne. For example
+
+**Team** model has the `playerId` key which makes it **target** relation. We should use the HasOne
+
+```js
+Player.hasOne(Team); // This will add `playerId` into Team model
```

Comment:
Not sure this accurately describes it, perhaps setup a case with Team, Coach and Player where Coach and Player are all types of Person.